server/route.go

```diff
@@ -158,7 +158,10 @@ func (c *client) processRouteInfo(info *Info) {
 		}
 		// If the server Info did not have these URLs, update and send an INFO
 		// protocol to all clients that support it (unless the feature is disabled).
-		if s.updateServerINFO(info.ClientConnectURLs) {
+		// Also send out the notification if the 'always notify' flag is on regardless
+		// of whether the server was already in the Server Info.
+		opts := s.getOpts()
```

Comment:
The function below does get the options, so I would move the check of always notify inside that function (that we may want to rename. The way it is now, you would send the notification even if the `NoAdvertise` boolean was set to false, which I don't think is right. By moving the test of AlwaysNotify at the end of the `updateServerINFO()` function, you would not by-pass the `NoAdvertise` flag.
Some additional tests would also be welcomed.

Reply:
Agreed.

I will move the check into the underlying updateServerINFO function so the NoAdvertise is not overwritten.